Extension/src/main.ts

```diff
@@ -138,27 +138,30 @@ export function activate(context: vscode.ExtensionContext): void {
     registerTempCommand("C_Cpp.TakeSurvey");
 
     processRuntimeDependencies(() => {
-        downloadCpptoolsJsonPkg().then(() => {
-            util.readFileText(util.getExtensionFilePath("cpptools.json"))
-                .then((cpptoolsString) => {
-                    processCpptoolsJson(cpptoolsString);
-                })
-                .catch((error) => {
-                    // We already log telemetry if cpptools.json fails to download.
-                })
-                .then(() => {
-                    // Main activation code.
-                    tempCommands.forEach((command) => {
-                        command.dispose();
-                    });
-                    tempCommands = [];
-                    LanguageServer.activate(delayedCommandsToExecute);
-                    delayedCommandsToExecute.forEach((command) => {
-                        vscode.commands.executeCommand(command);
-                    });
-                    delayedCommandsToExecute.clear();
+        util.readFileText(util.getExtensionFilePath("cpptools.json"))
```

Comment:
Can you check and make sure the file exists before you read it?

Reply:
That shouldn't be necessary. The file should always be there unless someone manually deletes it or something else impossible happens and we catch any exceptions in that case too.

Reply:
But we shouldn't fail activation if the file isn't there, we should activate with baseline values. why should the file missing make activation stop?

Reply:
We don't. There's a catch, right?

Reply:
If Andrew wants to check the file's existence in his refactor pull request, I'd be fine with that.